### ISSUE TITLE
 #define E1_AUTO_FAN_PIN

### ISSUE THREAD

**Issue by dimaknight1109** [bigtreetech/BIGTREETECH-SKR-V1.3#532]
I would like to add my hotend fan to the following which can be found in configuration_adv so that the fan turns off when its lower than 55 degree and turns on which it reaches 55.

#define E1_AUTO_FAN_PIN

How do i make this work. what do i need to type after "#define E1_AUTO_FAN_PIN"

**Comment by dimaknight1109** [comment]
#define E1_AUTO_FAN_PIN  P2_04 would this be correct?

**Comment by radek8** [comment]
You didn't write what kind of board you had

**Comment by hopeinformer** [comment]
It's easier to use the E0 rather than E1 for the fan, in which case this is true
#define E0_AUTO_FAN_PIN P2_04
That's the one that works for the SKR v1.4 & the Turbo, then uncomment and set the value in the line below

#define EXTRUDER_AUTO_FAN_TEMPERATURE 50, or in your case "55"

But radek8 is right, this pin will depend on which board you're using. Just look up the schematics and pins for your board to find the right one.
*Note: the pins are polarized. Be sure you have the polarity correct or your fan will spin in the wrong direction.

![Screenshot_20210623-073328_Gallery](https://user-images.githubusercontent.com/52873265/123117135-5962e100-d3f6-11eb-8364-bc2d900fea03.png)

![E0 E1 fan pins](https://user-images.githubusercontent.com/52873265/123116879-23bdf800-d3f6-11eb-84b8-1e89ae344f7a.png)

**Comment by dimaknight1109** [comment]
now I'm confused, lol
I read on some documentation i found online that i need to do the following

define E0_AUTO_FAN_PIN FAN1_PIN

Does not mention anything about adding a pin number.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by radek8** [comment]
You have to define the pin as @hopeinformer writes.
If you don't define a pin, Marlin can't start the fan because it doesn't know where it's connected

**Comment by dimaknight1109** [comment]
I get that but on the documents that can with bigtreetech they does state
you need to add a pin number. Reason why I'm confused.

**Comment by radek8** [comment]
Yes, the SKRV1.3 board does not have a dedicated connector or MOSFET for the extruder fan.
The connector for heating the second hotend is unused (unless you use two hotends) and can be redefined to connect the fan  extruder cooler.
What's incomprehensible about that?

**Comment by dimaknight1109** [comment]
This is what the document states to use the hotend fan
define E0_AUTO_FAN_PIN FAN1_PIN

This is what I pickup on when watching a YouTube video
#define E0_AUTO_FAN_PIN P2_04

This is my confusion. Which one is right?

**Comment by radek8** [comment]
Thirdly, I say that this is correct
#define E0_AUTO_FAN_PIN P2_04

You must replace "FAN1_PIN" with the pin number to which you connect the fan.

Connect the fan to the connector as shown above

**Comment by radek8** [comment]
You can also turn off and control the fan speed, which cools the control board with the motor drivers, but you would have to buy an external Mosfet to switch the fan.

**Comment by radek8** [comment]
It would not be the first mistake in Bigtreetech's documentation.
Bigtreetech states that the fan is connected to the pins of the FAN1 connector
But this connector has a continuous power supply.

![image](https://user-images.githubusercontent.com/46979052/123171102-a4730780-d47b-11eb-97df-5ce89893c7cb.png)

**Comment by dimaknight1109** [comment]
So it's #define E0_AUTO_FAN_PIN P2_04

**Comment by dimaknight1109** [comment]
apologies have one more question,
For dual Z, im using the E1 motor. I have defined Dual Z in Configuration
_adv. but for defining the stepper motors. do i add TMC2208:

#define E1_DRIVER_TYPE TMC2208_STANDALONE
or
#define Z2_DRIVER_TYPE TMC2208_STANDALONE

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by radek8** [comment]
What is your reason for using STANDALONE mode?
SKR boards are prepared for UART mode which can be configured at runtime

**Comment by dimaknight1109** [comment]
I have seen videos on this but dont know much about what to do for each
modes. Standalone seem to be the easiest.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by radek8** [comment]
Watch instructional videos about UART mode :-)

**Comment by dimaknight1109** [comment]
sure ill check som out, but what the difference with UART and standalone

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by radek8** [comment]
Monitor Trinamic drivers
   * for error conditions like overtemperature and short to ground.
   * To manage over-temp Marlin can decrease the driver current until the error condition clears.
   * Other detected conditions can be used to stop the current print.
   * Relevant G-codes:
   * M906 - Set or get motor current in milliamps using axis codes X, Y, Z, E. Report values if no axis codes given.
   * M911 - Report stepper driver overtemperature pre-warn condition.
   * M912 - Clear stepper driver overtemperature pre-warn condition flag.
   * M122 - Report driver parameters (Requires TMC_DEBUG)
   * M913 - TMC Hybrid Threshold Speed
   * M569 - TMC StealthChop
